### PR TITLE
RD-1066 Implement Tooltip

### PR DESCRIPTION
**Pull request by oznakn**
[loom video](https://www.loom.com/share/05f50297505b4f58a47321b58613e87f)

### Notes

I am aware that there is a problem with colors, Dang and I are trying to solve that issue.

### CODE REVIEW

File: src/components/common/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+export enum TooltipPosition {
+  top = 'top',
+  bottom = 'bottom',
+  left = 'left',
+  right = 'right',
+}
+
+export interface TooltipProps {
+  className?: string;
+  title: string;
```
Comment (danny-dang) [inline src/components/common/Tooltip/Tooltip.tsx:13]: perhaps `title` can be `string | ReactNode`. Sometimes, we may want to have more control over the styling for the title like this:

```
This text is <strong>bold</strong>
```

**Review by thellimist — COMMENTED** [review]
should this component be smart enough to figure out top/bottom/left/right depending on where it is on the page?

**Comment by danny-dang** [comment]
> should this component be smart enough to figure out top/bottom/left/right depending on where it is on the page?

Sometimes we also want it be the exact position. Maybe by default we can make it automatically calculate the position, and we can also override the position by the parent props. 

I have a function in the MenuDropdown component where it  calculate the position. you can check it. @oznakn

**Comment by oznakn** [comment]
Hey, here is the auto position logic added [loom video](https://www.loom.com/share/21cd5a150f7047bd8747a81266e4360e). @danny-dang @thellimist

**Comment by oznakn** [comment]
> Can we put components and other stuff inside the tooltip or is it only text?
> 
> Putting links is a common behaviour we have been using in tooltips

it can be anything

**Review by thellimist — APPROVED** [review]

**Pull request merged**